### PR TITLE
Leverage stoploss

### PR DESCRIPTION
**Pull request by samgermain**
Stoploss accounts for leverage

----------------------

A long trade that opens 
- at a price of `1.0` pair/USDT
- **with 5.0 leverage**
- a stoploss percent of 0.05

Would have a stoploss value of 0.99 pair/USDT

-------------------

A long trade that opens 
- at a price of `1.0` pair/USDT
- **with 1.0 leverage**
- a stoploss percent of 0.05

Would have a stoploss value of 0.95 pair/USDT

### CODE REVIEW

File: tests/test_freqtradebot.py [hunk 1]
```diff
@@ -1128,6 +1128,9 @@ def test_handle_stoploss_on_exchange(mocker, default_conf_usdt, fee, caplog, is_
     assert trade.is_open is False
     caplog.clear()
 
+    # TODO-lev: Test 2 identical orders but one with leverage, 1 without, and test that the
+    # leveraged trade is hit, but the other trade is not
```
Comment (samgermain) [inline tests/test_freqtradebot.py]: Why are we expecting the stoploss to be hit in the code a few lines above this?

Comment (xmatthias) [reply tests/test_freqtradebot.py]: it's a stoploss on exchange - so the exchange returns "status=closed"  when asking for the order.

Price didn't have to move - and i'm not sure we must adapt this part for leverage either, as the exchange will still raise the stop when necessary (what we should test is that the value we place it is properly calculated)

File: freqtrade/freqtradebot.py [hunk 2]
```diff
@@ -1065,7 +1065,11 @@ def handle_stoploss_on_exchange(self, trade: Trade) -> bool:
 
         # If enter order is fulfilled but there is no stoploss, we add a stoploss on exchange
         if not stoploss_order:
-            stoploss = self.edge.stoploss(pair=trade.pair) if self.edge else self.strategy.stoploss
+            stoploss = (
+                self.edge.stoploss(pair=trade.pair)
+                if self.edge else
+                self.strategy.stoploss / trade.leverage
+            )
```
Comment (samgermain) [inline freqtrade/freqtradebot.py:1072]: Comment for the link

File: tests/test_freqtradebot.py (hunk 1 above)
Comment (samgermain) [reply tests/test_freqtradebot.py]: I was just trying to test [these changes](https://github.com/freqtrade/freqtrade/pull/6466/files#r815318556), but given what you said I don't think that would be tested here

**Comment by samgermain** [comment]
This failing test keeps showing up in random builds across different PRs 

`FAILED tests/rpc/test_rpc_apiserver.py::test_api_trade_single[True] - assert ...`

It didn't happen locally when running `pytest --random-order-bucket=module --random-order-seed=560592` with python3.9

**Comment by xmatthias** [comment]
> This failing test keeps showing up in random builds across different PRs
> 
> `FAILED tests/rpc/test_rpc_apiserver.py::test_api_trade_single[True] - assert ...`

"should" be fixed in feat/short - it's an odd random failure i've got no clue why it happens.

**Comment by samgermain** [comment]
> > This failing test keeps showing up in random builds across different PRs
> > `FAILED tests/rpc/test_rpc_apiserver.py::test_api_trade_single[True] - assert ...`
> 
> "should" be fixed in feat/short - it's an odd random failure i've got no clue why it happens.

K, I rebased, in case you pulled

**Comment by samgermain** [comment]
> > This failing test keeps showing up in random builds across different PRs
> > `FAILED tests/rpc/test_rpc_apiserver.py::test_api_trade_single[True] - assert ...`
> 
> "should" be fixed in feat/short - it's an odd random failure i've got no clue why it happens.

It looks like it's still happening

File: freqtrade/freqtradebot.py (hunk 2 above)
Comment (xmatthias) [reply freqtrade/freqtradebot.py:1072]: which link?
ah nevermind

**Comment by xmatthias** [comment]
> It looks like it's still happening

just need to forcepush harder :rofl: 
seems like i got it now though

File: freqtrade/freqtradebot.py (hunk 2 above)
Comment (xmatthias) [reply freqtrade/freqtradebot.py:1072]: i actually think i would've missed this ... :+1:

File: tests/test_freqtradebot.py (hunk 1 above)
Comment (xmatthias) [reply tests/test_freqtradebot.py]: there's a difference - the changes are fine and will change how/where the stoploss will be placed.
That'll not change the "stop on exchange" handling though

**Review by xmatthias — APPROVED** [review]
Nice one :+1:
